source/extensions/filters/http/rate_limit_quota/quota_bucket_cache.h

```diff
@@ -41,13 +41,16 @@ struct Bucket {
   // RLQS server.
   BucketId bucket_id;
   // Cached action from the response that was received from the RLQS server.
-  BucketAction bucket_action;
+  absl::optional<BucketAction> cached_action = absl::nullopt;
+  BucketAction default_action;
   // Cache quota usage.
   QuotaUsage quota_usage;
   // Rate limiter based on token bucket algorithm.
   TokenBucketPtr token_bucket_limiter;
   // First assignment time.
-  Envoy::MonotonicTime first_assignment_time = {};
+  Envoy::MonotonicTime first_assignment_time;
+  // Most recent assignment time.
```

Comment:
nit: Do we really need this new variable? Can we just use `first_assignment_time`?

But it is up to you.

Reply:
If anything I'd delete `first_assignment_time` as we don't use that anywhere. It's not accurate to what we need to calculate expiration (thus the new variable).

Reply:
Yeah, if we actually only need one variable let's just keep one and rename it to make it accurate.

Reply:
The var has been cleaned up.